go/controller/api/v1alpha1/toolserver_types.go

```diff
@@ -31,17 +31,54 @@ type ToolServerConfig struct {
 	Sse   *SseMcpServerConfig   `json:"sse,omitempty"`
 }
 
+// ValueSource defines a source for configuration values from a Secret or ConfigMap
+type ValueSource struct {
+	Name      string `json:"name"`
+	Key       string `json:"key"`
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
Everywhere else in the codebase we're following an implicit `ref` practice rather than explicit `name` + `namespace`. Meaning that to indicate a namespace, it should be separated with a `/`. e.g. `name/namespace